applets/wncklet/window-list.c

```diff
@@ -940,169 +727,118 @@ static void display_about_dialog(GtkAction* action, TasklistData* tasklist)
 		NULL);
 }
 
-static void group_windows_toggled(GtkToggleButton* button, TasklistData* tasklist)
+static void mouse_scrolling_callback(GSettings* settings, gchar *key, TasklistData* tasklist)
 {
-	if (gtk_toggle_button_get_active(button))
-	{
-		gchar *value;
-		value = g_object_get_data (G_OBJECT (button), "group_value");
-		g_settings_set_string (tasklist->settings, "group-windows", value);
-	}
+	wnck_tasklist_set_scroll_enabled (WNCK_TASKLIST(tasklist->tasklist), g_settings_get_boolean (settings, "scroll-enabled"));
 }
 
-#ifdef HAVE_WINDOW_PREVIEWS
-static void thumbnail_size_spin_changed(GtkSpinButton* button, TasklistData* tasklist)
+static void move_unminimized_windows_callback(GSettings* settings, gchar *key, TasklistData* tasklist)
 {
-	g_settings_set_int(tasklist->preview_settings, "thumbnail-window-size", gtk_spin_button_get_value_as_int(button));
+	wnck_tasklist_set_switch_workspace_on_unminimize(WNCK_TASKLIST(tasklist->tasklist), g_settings_get_boolean (settings, "move-unminimized-windows"));
 }
-#endif
 
-static void move_minimized_toggled(GtkToggleButton* button, TasklistData* tasklist)
+static void display_all_workspaces_callback(GSettings* settings, gchar *key, TasklistData* tasklist)
 {
-	g_settings_set_boolean(tasklist->settings, "move-unminimized-windows", gtk_toggle_button_get_active(button));
+	wnck_tasklist_set_include_all_workspaces(WNCK_TASKLIST(tasklist->tasklist), g_settings_get_boolean (settings, "display-all-workspaces"));
 }
 
-static void display_all_workspaces_toggled(GtkToggleButton* button, TasklistData* tasklist)
+#ifdef HAVE_WINDOW_PREVIEWS
+static void on_thumbnail_size_spin_value_changed(GtkSpinButton* button, TasklistData* tasklist)
 {
-	g_settings_set_boolean(tasklist->settings, "display-all-workspaces", gtk_toggle_button_get_active(button));
+	g_settings_set_int(tasklist->preview_settings, "thumbnail-window-size", gtk_spin_button_get_value_as_int(button));
 }
+#endif
 
-#define WID(s) GTK_WIDGET(gtk_builder_get_object(builder, s))
-
-static void setup_sensitivity(TasklistData* tasklist, GtkBuilder* builder, const char* wid1, const char* wid2, const char* wid3, const char* key)
+static void on_never_group_radio_toggled(GtkRadioButton* button, TasklistData* tasklist)
 {
-	GtkWidget* w;
-
-	if (g_settings_is_writable(tasklist->settings, key))
-	{
-		return;
-	}
-
-	w = WID(wid1);
-	g_assert(w != NULL);
-	gtk_widget_set_sensitive(w, FALSE);
-
-	if (wid2 != NULL)
-	{
-		w = WID(wid2);
-		g_assert(w != NULL);
-		gtk_widget_set_sensitive(w, FALSE);
-	}
-
-	if (wid3 != NULL)
-	{
-		w = WID(wid3);
-		g_assert(w != NULL);
-		gtk_widget_set_sensitive(w, FALSE);
-	}
+	g_settings_set_enum (tasklist->settings, "group-windows", 0);
```

Comment:
Using the macro 
WNCK_TASKLIST_NEVER_GROUP replaces the number 0
WNCK_TASKLIST_AUTO_GROUP                                        1
WNCK_TASKLIST_ALWAYS_GROUP                                    2

Reply:
done. I use a custom enum though, because we cant be sure that the enum from libwnck always matches our code.